kivymd/uix/button.py

```diff
@@ -895,7 +895,7 @@ def on_text_color(self, instance, value):
             self._current_text_color = value
 
     def on_md_bg_color(self, instance, value):
-        if value != self.theme_cls.primary_color:
+        if value != self._current_button_color:
             self._current_button_color = value
 
```

Comment:
This code does not solve any bug, since it will make the button lose its current color and be replaced by the theme assigned one.

Reply:
Huh? I don't understand. My PR fixes the scenario where this works:
```
button = MDIconButton(text="OK")
button.md_bg_color = app.theme_cls.primary_color
# do something
button.text = "ERROR"
button.md_mg_color = app.theme_cls.error_color
```
But this doesn't:
```
button = MDIconButton(text="CANCEL")
button.md_bg_color = app.theme_cls.error_color
# do something
button.text = "OK"
button.md_mg_color = app.theme_cls.primary_color
```

Reply:
Try changing the theme of the app, and you'll see the problem, as well others too.

Ideally, you should be able to assign a theme for the buttons, so the app will update all your UI to your theme color scheme.

For example, you should be able to set button.theme_button_color = "Error", so it will change it's color, internally.
And the md_bg_color won't interfere with the actual modification.

there's already a pr that solves this and other problems with the buttons

check #501 for more info